### DISCUSSION TITLE
Language codes in OWSM models

### DISCUSSION THREAD

**Discussion by alumae** [espnet/espnet#5529] [General]
I am confused about the language codes in OWSM models. The token list in the OWSM models on Huggingface has three letter codes (`<eng>", "<deu>`), but data preparation scripts in under `egs2/mixed_v2/` (e.g. [this](https://github.com/espnet/espnet/blob/master/egs2/mixed_v1/s2t1/local/prepare_spgispeech.py)) use two letter codes (`<en>`, etc). Can you clarify this? Which one should we use when finetuning OWSM model on custom data?

**Comment by sw005320** [comment]
@pyf98, can you answer it?

**Comment by pyf98** [comment]
Hi, thanks for the question. We used two letter codes for v1 and v2. Then we switched to three letter codes for v3 in order to support more languages. You can find the token vocabulary in each model (you can see it in the config file in an uploaded model on Hugging Face)
